lib/server/websocket.js

```diff
@@ -0,0 +1,525 @@
+          var queryDev;
+          if (data.data.NSCLIENT_ID) {
+            queryDev = { NSCLIENT_ID: data.data.NSCLIENT_ID };
+          } else {
+            queryDev = {
+              created_at: data.data.created_at
+            };
+          }
+
+          // try to find exact match
+          ctx.store.collection(collection).find(queryDev).toArray(function findResult (err, array) {
+            if (err || array.length > 0) {
+              console.log(LOG_DEDUP + 'Devicestatus exact match');
+              if (callback) {
+                callback([array[0]]);
+              }
+              return;
+            }
+          });
+          ctx.store.collection(collection).insert(data.data, function insertResult (err, doc) {
+            if (err != null && err.message) {
+              console.log('devicestatus insertion error: ', err.message);
+              return;
+            }
+            if (callback) {
+            callback(doc.ops);
+            }
+            ctx.bus.emit('data-received');
+          });
+        } else {
+          ctx.store.collection(collection).insert(data.data, function insertResult (err, doc) {
+            if (err != null && err.message) {
+              console.log(data.collection + ' insertion error: ', err.message);
+              return;
+            }
+            if (callback) {
+              callback(doc.ops);
+            }
+            ctx.bus.emit('data-received');
+          });
+        }
+      });
+      // dbRemove message
+      //  {
+      //    collection: treatments
+      //    _id: 'some mongo record id'
+      //  }
+      socket.on('dbRemove', function dbRemove (data, callback) {
+       console.log(LOG_WS + 'dbRemove client ID: ', socket.client.id, ' data: ', data);
+       var collection = supportedCollections[data.collection];
+
+        var check = checkConditions('dbUpdate', data);
+        if (check) {
+         if (callback) {
+            callback( check );
+          }
+          return;
+        }
+
+        var objId = new ObjectID(data._id);
+        ctx.store.collection(collection).remove(
+          { '_id': objId }
+        );
+
+        if (callback) {
+          callback( { result: 'success' } );
+        }
+        ctx.bus.emit('data-received');
+      });
+
+      // Authorization message
+      // {
+      //  client: 'web' | 'phone' | 'pump'
+      //  , secret: 'secret_hash'
+      //  [, history : history_in_hours ]
+      //  [, status : true ]
+      // }
+      socket.on('authorize', function authorize (message, callback) {
+        verifyAuthorization(message, function verified (err, authorization) {
+          socketAuthorization = authorization;
+          clientType = message.client;
+          history = message.history || 48; //default history is 48 hours
+          var from = message.from;
+
+          if (socketAuthorization.read) {
+            socket.join('DataReceivers');
+            var filterTreatments = false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterTreatments' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/m-kozak/cgm-remote-monitor/pullRequest?prid=1196151)